python/tvm/build_module.py

```diff
@@ -52,17 +57,58 @@ def __getattr__(self, name):
             return BuildConfig.defaults[name]
         return self._attr[name]
 
+    def decorator(self, func):
+        ''' decorate the pass function'''
+        def dump(*args, **kwargs):
+            '''dump function'''
+            retv = func(*args, **kwargs)
+            # pylint: disable=unidiomatic-typecheck
+            if type(retv) not in self._types:
+                return retv
+            pname = str(self._pass_id) + "_" + func.func_name + "_ir.cc"
+            with open(pname, "a") as f:
+                out = retv.body if isinstance(retv, container.LoweredFunc) else retv
+                print(out, file=f)
+                if isinstance(retv, container.Array):
+                    for x in retv:
+                        out = x.body if isinstance(x, container.LoweredFunc) else x
+                        print("---------", x.name, "\n", out, "\n-----------\n", file=f)
+                self._pass_id += 1
+            return retv
+        return dump
+
+    def decorate_irpass(self):
+        for k, v in vars(ir_pass).items():
+            vars(ir_pass)[k] = self.decorator(v) if isinstance(v, types.FunctionType) else v
```

Comment:
call vars in the beginning to get the dict, them use the dict over time.

Reply:
I've considered that, what i'm not sure is, if i use a dict to manage function list just as add_lower_pass, all the ir_pass function calls in lower, build and in testcases would be affected, is this avoidable or acceptable?